packages/core/src/api/thread.ts

```diff
@@ -0,0 +1,121 @@
+import type { RawFile, REST } from '@discordjs/rest';
+import {
+	Routes,
+	type APIThreadChannel,
+	type APIThreadMember,
+	type RESTPostAPIChannelThreadsJSONBody,
+	type RESTPostAPIGuildForumThreadsJSONBody,
+} from 'discord-api-types/v10';
+
+export interface StartThreadOptions extends RESTPostAPIChannelThreadsJSONBody {
+	message_id?: string;
+}
+
+export interface StartForumThreadOptions extends RESTPostAPIGuildForumThreadsJSONBody {
+	message: RESTPostAPIGuildForumThreadsJSONBody['message'] & { files?: RawFile[] };
+}
+
+export class ThreadsAPI {
+	public constructor(private readonly rest: REST) {}
+
+	/**
+	 * Fetches a thread
+	 *
+	 * @param channelId - The id of the channel to fetch the thread from
+	 * @param threadId - The id of the thread
+	 */
+	public async get(channelId: string, threadId: string) {
+		return (await this.rest.get(Routes.threads(channelId, threadId))) as APIThreadChannel;
+	}
+
+	/**
+	 * Creates a new thread
+	 *
+	 * @param channelId - The id of the channel to start the thread in
+	 * @param options - The options to use when starting the thread
+	 */
+	public async start(channelId: string, options: StartThreadOptions) {
+		const { message_id, ...body } = options;
+		return (await this.rest.post(Routes.threads(channelId, message_id), {
+			body,
+		})) as APIThreadChannel;
+	}
+
+	/**
+	 * Creates a new forum post
+	 *
+	 * @param channelId - The id of the forum channel to start the thread in
+	 * @param options - The options to use when starting the thread
+	 */
+	public async startForumThread(channelId: string, options: StartForumThreadOptions) {
+		const { message, ...optionsBody } = options;
+		const { files, ...messageBody } = message;
```

Comment:
```suggestion
	public async startForumThread(channelId: string, { message, ...optionsBody }: StartForumThreadOptions) {
		const { files, ...messageBody } = message;
```

PS: This could all be done in the parameters if you really want to